torch_xla/csrc/ops/randperm.cpp

```diff
@@ -0,0 +1,73 @@
+#include "torch_xla/csrc/ops/randperm.h"
+
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+#include "torch_xla/csrc/ops/xla_ops.h"
+#include "tsl/platform/stacktrace.h"
+#include "xla/client/lib/loops.h"
+#include "xla/shape_util.h"
+
+namespace torch_xla {
+namespace {
+
+using namespace xla;
+
+xla::Shape NodeOutputShape(int64_t n) {
+  return xla::ShapeUtil::MakeShape(xla::PrimitiveType::S64, {n});
+}
+
+XlaOp Swap(XlaOp input, XlaOp i, XlaOp j) {
+  XlaOp i_value = xla::DynamicSlice(input, {i}, /*slice_sizes=*/{1});
+  XlaOp j_value = xla::DynamicSlice(input, {j}, /*slice_sizes=*/{1});
+
+  XlaOp write_i = xla::DynamicUpdateSlice(input, j_value, {i});
+  XlaOp write_j = xla::DynamicUpdateSlice(write_i, i_value, {j});
+
+  return write_j;
+}
+
+StatusOr<std::vector<XlaOp>> LoopBodyFn(XlaOp i, absl::Span<const XlaOp> values,
+                                        XlaBuilder* builder) {
+  XlaOp input_array = values[0];
+  XlaOp upper_bound_exclusive = values[1];
+
+  XlaOp target_index = xla::RngUniform(
+      i, upper_bound_exclusive,
+      ShapeUtil::MakeShape(xla::PrimitiveType::S64, /*dimensions=*/{1}));
+
+  XlaOp swapped_array = Swap(input_array, i, target_index);
+  return std::vector<XlaOp>{swapped_array, upper_bound_exclusive};
+}
+
+}  // namespace
+
+RandPerm::RandPerm(int64_t n, const at::ScalarType dtype,
+                   const at::Layout layout, const at::Device device,
+                   bool pin_memory)
+    : XlaNode(
+          torch::lazy::OpKind(at::aten::randperm), /*operands=*/{},
+          [&]() { return NodeOutputShape(n); }, /*num_outputs=*/1,
+          torch::lazy::MHash(n)),
```

Comment:
I think I had to do this to prevent the compilation from being cached all the time, but I hope you can double check to make sure this is correct :). Thanks!